flink-doris-connector/src/main/java/org/apache/doris/flink/source/reader/DorisSourceSplitReader.java

```diff
@@ -69,8 +75,17 @@ private void checkSplitOrStartNext() throws IOException {
             throw new IOException("Cannot fetch from another split - no split remaining");
         }
         currentSplitId = nextSplit.splitId();
-        valueReader =
-                new DorisValueReader(nextSplit.getPartitionDefinition(), options, readOptions);
+        if (readOptions.getUseFlightSql()) {
+            valueReader =
+                    new DorisFlightValueReader(
+                            nextSplit.getPartitionDefinition(),
+                            options,
+                            readOptions,
+                            RestService.getSchema(options, readOptions, LOG));
```

Comment:
Can this be put into the init of DorisFlightValueReader so that other places can call it without worrying about it?